### PR TITLE
Migrate log4j2 update.

### PR DESCRIPTION
**Pull request by hasinidilanka**
### Proposed changes in this pull request

- Remove carbon.logging and commons-logging dependency.
- Add pax-logging dependency
- Remove maven-scr plugin

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -357,18 +373,18 @@
         <!-- Identity Agent Entitlement proxy version -->
         <identity.agent.entitlement.proxy.version>${project.version}</identity.agent.entitlement.proxy.version>
         <identity.agent.entitlement.proxy.exp.pkg.version>${identity.agent.entitlement.proxy.version}</identity.agent.entitlement.proxy.exp.pkg.version>
-        <identity.agent.entitlement.proxy.imp.pkg.version.range>[5.0.0,6.0.0)</identity.agent.entitlement.proxy.imp.pkg.version.range>
+        <identity.agent.entitlement.proxy.imp.pkg.version.range>[5.3.0,6.0.0)</identity.agent.entitlement.proxy.imp.pkg.version.range>
```
Comment (tharindu-bandara) [inline pom.xml]: add space after `,`

File: pom.xml [hunk 2]
```diff
@@ -377,9 +393,8 @@
         <commons-lang.wso2.version>2.6.0.wso2v1</commons-lang.wso2.version>
         <commons-io.wso2.version>2.4.0.wso2v1</commons-io.wso2.version>
         <commons-httpclient.wso2.version>3.1.0.wso2v2</commons-httpclient.wso2.version>
-        <commons-logging.version>1.2</commons-logging.version>
         <commons-httpclient.wso2.osgi.version>[3.1.0,4.0.0)</commons-httpclient.wso2.osgi.version>
-        <commons-logging.osgi.version.range>[1.2,2.0)</commons-logging.osgi.version.range>
+        <commons-logging.osgi.version.range>[1.2.0,2.0.0)</commons-logging.osgi.version.range>
```
Comment (tharindu-bandara) [inline pom.xml]: add space after the comma

File: pom.xml (hunk 1 above)
Comment (hasinidilanka) [reply pom.xml]: Fixed - https://github.com/wso2-extensions/identity-agent-entitlement-proxy/pull/30

File: pom.xml (hunk 2 above)
Comment (hasinidilanka) [reply pom.xml]: Fixed - https://github.com/wso2-extensions/identity-agent-entitlement-proxy/pull/30

**Pull request merged**